README.md

```diff
@@ -1,5 +1,189 @@
-# node-core
+# node-core 🧬
 
 Core libraries for Node.js backend services.
 
-See [docs](/docs) for instructions on how to use.
+* [Overview](#overview)  
+* [HTTP Client](#http-client)  
+* [Default Logging Configuration](#default-logging-configuration)  
+* [ConfigScope](#configscope)  
+* [Error Handling](#error-handling)  
+
+See [docs](/docs) for further instructions on how to use.
+
+## Overview
+
+### Dependencies
+
+* `pino`;
+* `unidici`;
+* `unidici-retry`.
+
+## HTTP Client
+
+The library provides methods to implement the client side of HTTP protocols. Public methods available are:
+
+* `buildClient()`, which returns a [Client](https://undici.nodejs.org/#/docs/api/Client) instance and should be called before any of the following methods with parameters:
+    * `baseUrl`;
+    * `clientOptions` – set of [ClientOptions](https://undici.nodejs.org/#/docs/api/Client?id=parameter-clientoptions) (optional). If none are provided, the following default options will be used to instantiate the client:
+        ```ts
+        keepAliveMaxTimeout: 300_000,
+        keepAliveTimeout: 4000,
+        ```
+* `sendGet()`;
+* `sendPost()`;
+* `sendPut()`;
+* `sendPutBinary()`;
+* `sendDelete()`;
+* `sendPatch()`.
+
+All _send_ methods accept a type parameter and the following arguments:
+* `client`, the return value of `buildClient()`;
+* `path`;
+* `options` – (optional). Possible values are:
+    * `headers`;
+    * `query`;
+    * `timeout`;
+    * `throwOnError`;`
+    * `reqContext`;
+    * `safeParseJson`;
+    * `disableKeepAlive`;`
+    * `retryConfig`, defined by:
+        * `maxAttempts`, the maximum number of times a request should be retried;
+        * `delayBetweenAttemptsInMsecs`;
+        * `statusCodesToRetry`, the status codes that trigger a retry;
+        * `retryOnTimeout`;
+    * `clientOptions`;
+    * `responseSchema`;
```

Comment:
explain that it is used both for inferring the response type of the call, and also (if validateResponse is true) for validating the response schema.